tests/playwright/src/specs/kind.spec.ts

```diff
@@ -121,21 +128,43 @@ describe('Kind End-to-End Tests', async () => {
       );
       test('Check resources added with the Kind cluster', async () => {
         const containersPage = await navigationBar.openContainers();
-        await playExpect.poll(async () => containersPage.containerExists(CONTAINER_NAME)).toBeTruthy();
-        const containerDetailsPage = await containersPage.openContainersDetails(CONTAINER_NAME);
+        await playExpect.poll(async () => containersPage.containerExists(KIND_CONTAINER_NAME)).toBeTruthy();
+        const containerDetailsPage = await containersPage.openContainersDetails(KIND_CONTAINER_NAME);
         await playExpect.poll(async () => await containerDetailsPage.getState()).toEqual(ContainerState.Running);
 
         const volumesPage = new VolumesPage(page);
-        const volumeName = await getVolumeNameForContainer(page, CONTAINER_NAME);
+        const volumeName = await getVolumeNameForContainer(page, KIND_CONTAINER_NAME);
         if (!volumeName) {
-          throw new Error(`Volume name for container "${CONTAINER_NAME}" is not defined.`);
+          throw new Error(`Volume name for container "${KIND_CONTAINER_NAME}" is not defined.`);
         }
         const volumeDetailsPage = await volumesPage.openVolumeDetails(volumeName);
         await playExpect.poll(async () => await volumeDetailsPage.isUsed()).toBeTruthy();
       });
       test('Validate correct Kubernetes context is selected', async () => {
         await statusBar.validateKubernetesContext(KUBERNETES_CONTEXT);
       });
+      describe('Deploy a container to the Kind cluster', async () => {
+        test('Pull an image and start a container', async () => {
+          const imagesPage = await navigationBar.openImages();
+          const pullImagePage = await imagesPage.openPullImage();
+          await pullImagePage.pullImage(IMAGE_TO_PULL);
+          await imagesPage.waitForImageExists(IMAGE_NAME, 10000);
+          await imagesPage.startContainerWithImage(IMAGE_NAME, CONTAINER_NAME);
+        });
+        test('Deploy the container ', async () => {
+          const containerDetailsPage = new ContainerDetailsPage(page, CONTAINER_NAME);
+          await playExpect(containerDetailsPage.heading).toBeVisible();
+          await playExpect(containerDetailsPage.deployButton).toBeVisible();
+          await containerDetailsPage.deployButton.click();
+          const deployToKubernetesPage = new DeployToKubernetesPage(page);
+          await deployToKubernetesPage.deployPod(CONTAINER_NAME, KUBERNETES_CONTEXT);
+        });
+        test('Verify the deployed application', async () => {
+          const podPage = await navigationBar.openPods();
+          console.log(DEPLOYED_POD_NAME);
+          await playExpect.poll(async () => podPage.deployedPodExists(DEPLOYED_POD_NAME)).toBeTruthy();
+        });
+      });
```

Comment:
I don't really understand why validating the pod creation needs to be its own test, why not check that at the end of the test that actually does the deploying?